packages/grafana-ui/src/components/Graph/GraphTooltip.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { stylesFactory, useTheme } from '../../themes';
+import { GrafanaTheme } from '../../types/theme';
+import { css, cx } from 'emotion';
+import { getTimeZoneDateFormatter, dateTime } from '@grafana/data';
+import { TooltipContentProps, TooltipMode } from '../Chart/Tooltip';
+import { SeriesIcon } from '../Legend/SeriesIcon';
+import { getMultiSeriesGraphHoverInfo } from './utils';
+
+export interface GraphTooltipOptions {
+  mode: TooltipMode;
+}
+
+const getGraphTooltipStyles = stylesFactory((theme: GrafanaTheme) => {
+  return {
+    seriesTable: css`
+      display: table;
+    `,
+    seriesTableRow: css`
+      display: table-row;
+    `,
+    seriesTableCell: css`
+      display: table-cell;
+    `,
+    value: css`
+      padding-left: ${theme.spacing.md};
+    `,
+    activeSeries: css`
+      font-weight: ${theme.typography.weight.bold};
+    `,
+  };
+});
+
+export const GraphTooltip: React.FC<TooltipContentProps> = ({
+  pos,
+  mode = 'single',
+  dimmensions,
+  activeDimmensions,
+  timeZone,
+}) => {
+  const theme = useTheme();
+  const styles = getGraphTooltipStyles(theme);
+  const dateFormatter = getTimeZoneDateFormatter(timeZone);
+
+  let content;
+
+  if (mode === 'single') {
+    // not hovering over a point, skip rendering
+    if (activeDimmensions.yAxis === undefined) {
+      return null;
+    }
+
+    // Assuming single x-axis, time
+    // Active dimension x-axis indicates a time field corresponding to y-axis value
+    const timeField = dimmensions['xAxis'][activeDimmensions.xAxis[0]];
```

Comment:
dimmensions['xAxis']  vs dimmensions.xAxis

I guess it is the same, but the latter looks a bit better